### ISSUE TITLE
Incorrect IntelliSense for functions with auto args

### ISSUE THREAD

**Issue by UndefBhvr** [microsoft/vscode-cpptools#6567]
Issue Type: <b>Bug</b>

The result type of  `read1()` is `void`, but IntelliSense marked it as `int`.

![屏幕截图 2020-11-23 154344](https://user-images.githubusercontent.com/46280451/99939874-06647500-2da6-11eb-916b-f8a5475788af.jpg)

In `main()` , the first time I tried to watch it, Intellisense worked correctly.
![屏幕截图 2020-11-23 154518](https://user-images.githubusercontent.com/46280451/99939854-fcdb0d00-2da5-11eb-862a-f0b92bb4b1bb.jpg)

But the second time I tried to watch it, IntelliSense failed.
![屏幕截图 2020-11-23 154430](https://user-images.githubusercontent.com/46280451/99939877-082e3880-2da6-11eb-99d1-439371421071.jpg)

Sometimes it looks like these:
![屏幕截图 2020-11-23 154625](https://user-images.githubusercontent.com/46280451/99939918-1e3bf900-2da6-11eb-90e6-d9229acc062e.jpg)
![屏幕截图 2020-11-23 154700](https://user-images.githubusercontent.com/46280451/99939922-2005bc80-2da6-11eb-97f4-3cbee305896e.jpg)

Code:
```cpp
#include<cstdio>
#include<cctype>

void _read(auto &x)
{
	x=0;
	char c=getchar();
	while(!isdigit(c))c=getchar();
	while(isdigit(c))x=x*10-48+c,c=getchar();
}

void read1()
{}

void read1(auto &x,auto &...args)
{
	_read(x);
	read1(args...);
}

// template<typename _Tp,typename ..._Args>
// void read1(_Tp &x,_Args &...args)
// {
// 	_read(x);
// 	read1(args...);
// }

int a,b,c;

int main()
{
	read1(a,b,c);
	printf("%d %d %d",a,b,c);
}
```

Extension version: 1.1.2
VS Code version: Code 1.51.1 (e5a624b788d92b8d34d1392e4c4d9789406efe8f, 2020-11-10T23:34:32.027Z)
OS version: Windows_NT x64 10.0.19041

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz (4 x 3192)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.91GB (2.53GB free)|
|Process Argv|--crash-reporter-id 607dc872-fe4f-48ca-b7ee-e540cba2be15 --crash-reporter-id 607dc872-fe4f-48ca-b7ee-e540cba2be15|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by Colengms** [comment]
Hi @ILSYT .  Thanks for reporting this.  This also repro's in VS, so I opened an issue there.  (We share code for IntelliSense with VS).  That issue is here: https://developercommunity.visualstudio.com/content/problem/1266761/incorrect-intellisense-for-functions-with-auto-arg.html

**Comment by sean-mcmanus** [comment]
This got fixed with 1.3.1.
